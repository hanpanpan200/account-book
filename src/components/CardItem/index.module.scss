.container {
  display: flex;
  margin: 10px 0;

  .left {
    display: flex;
    align-items: center;
    width: 50px;
  }
  .icon {
    width: 30px;
    height: 30px;
  }
  .middle {
    flex: 1;
  }
  .right {
    max-width: 120px;
    overflow: hidden;
    text-overflow: ellipsis;
    white-space: nowrap;
    text-align: right;
    font-weight: 500;
  }
  .text {
    color: #4a4c4e;
    font-size: 16px;
    line-height: 26px;
    height: 26px;
  }
  .light {
    font-size: 14px;
    color: #7f7f7f;
  }
}