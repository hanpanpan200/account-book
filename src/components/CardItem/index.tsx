import React from 'react';
import classNames from 'classnames/bind';

import styles from './index.module.scss';

const cx = classNames.bind(styles);

interface Props {
  iconSource: string
  title: string | React.ReactElement
  description: string | React.ReactElement
  content: string
}

const CardItem:React.FC<Props> = ({iconSource, title, description, content}) => {
  return (
    <div className={styles.container}>
      <div className={styles.left}>
        <img alt='leftIcon' className={styles.icon} src={iconSource} />
      </div>
      <div className={styles.middle}>
        <div className={styles.text}>{title}</div>
        <div className={cx(styles.text, styles.light)}>{description}</div>
      </div>
      <div className={cx(styles.right, styles.text)}>{content}</div>
    </div>
  )
}

export default CardItem;