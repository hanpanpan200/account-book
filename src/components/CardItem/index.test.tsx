import React from 'react';
import { render } from '@testing-library/react';
import CardItem from './index';

describe('CardItem', () => {
  it('should render correctly when title and description are provided as string', () => {
    const { container } = render(<CardItem iconSource='icon.svg' title='标题' description='描述' content='内容' />);
    expect(container).toMatchSnapshot();
  });
  it('should render correctly when title and description are provided as component', () => {
    const { container } =
      render(<CardItem iconSource='icon.svg' title={<div>hello</div>} description={<div>world</div>} content='内容' />);
    expect(container).toMatchSnapshot();
  });
});
