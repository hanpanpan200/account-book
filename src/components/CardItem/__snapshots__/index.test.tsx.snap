// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardItem should render correctly when title and description are provided as component 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="left"
    >
      <img
        alt="leftIcon"
        class="icon"
        src="icon.svg"
      />
    </div>
    <div
      class="middle"
    >
      <div
        class="text"
      >
        <div>
          hello
        </div>
      </div>
      <div
        class="text light"
      >
        <div>
          world
        </div>
      </div>
    </div>
    <div
      class="right text"
    >
      内容
    </div>
  </div>
</div>
`;

exports[`CardItem should render correctly when title and description are provided as string 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="left"
    >
      <img
        alt="leftIcon"
        class="icon"
        src="icon.svg"
      />
    </div>
    <div
      class="middle"
    >
      <div
        class="text"
      >
        标题
      </div>
      <div
        class="text light"
      >
        描述
      </div>
    </div>
    <div
      class="right text"
    >
      内容
    </div>
  </div>
</div>
`;
